macros/src/derive_fn.rs

```diff
@@ -136,15 +145,50 @@ pub fn derive_fn(
         #link_section
         pub static #spec_ident: [u8; #spec_xdr_len] = *#spec_xdr_lit;
 
-        #export_name
-        pub fn #wrap_ident(env: stellar_contract_sdk::Env, #(#wrap_args),*) -> stellar_contract_sdk::RawVal {
-            <_ as stellar_contract_sdk::IntoVal<stellar_contract_sdk::Env, stellar_contract_sdk::RawVal>>::into_val(
-                #call(
-                    #env_call
-                    #(#wrap_calls),*
-                ),
-                &env
-            )
+        pub mod #mod_ident {
+            #export_name
+            pub fn call(env: stellar_contract_sdk::Env, #(#wrap_args),*) -> stellar_contract_sdk::RawVal {
+                #use_trait;
+                <_ as stellar_contract_sdk::IntoVal<stellar_contract_sdk::Env, stellar_contract_sdk::RawVal>>::into_val(
+                    #call(
+                        #env_call
+                        #(#wrap_calls),*
+                    ),
+                    &env
+                )
+            }
+
+            pub fn call_slice(env: stellar_contract_sdk::Env, args: &[stellar_contract_sdk::RawVal]) -> stellar_contract_sdk::RawVal {
+                call(env, #(#slice_args),*)
+            }
+        }
+    }
+}
+
+#[allow(clippy::too_many_lines)]
+pub fn derive_add_functions<'a>(
+    ty: &Box<Type>,
+    methods: impl Iterator<Item = &'a syn::ImplItemMethod>,
+    feature: &Option<String>,
+) -> TokenStream2 {
+    let (idents, wrap_idents): (Vec<_>, Vec<_>) = methods
+        .map(|m| {
+            let ident = format!("{}", m.sig.ident);
+            let wrap_ident = format_ident!("__{}", m.sig.ident);
+            (ident, wrap_ident)
+        })
+        .multiunzip();
+    let cfg = if let Some(cfg_feature) = feature {
+        quote! { #[cfg(feature = #cfg_feature)] }
+    } else {
+        quote! {}
+    };
+    quote! {
+        #cfg
```

Comment:
I think it would be okay to hardcode `testutils` here. I think it will be helpful for the ecosystem if we encourage the testutils pattern, rather than make the macros highly customizable.

Reply:
It also might be good to also couple it with `tests`, so that developers writing tests automatically get access to the additional capabilities if they don't otherwise need to add the testutils feature for their test run, i.e.:
```rust
#[cfg(or(test, feature = "testutils"))]
```

Reply:
Opened https://github.com/stellar/rs-stellar-contract-sdk/issues/269